youths/schema/types.py

```diff
@@ -86,6 +95,36 @@ def resolve_renewable(self: YouthProfile, info, **kwargs):
             date.today()
         )
 
+    def resolve_profile(self: YouthProfile, info, **kwargs):
+        return self
```

Comment:
If I understand this bit correctly, this should be returning the `YouthProfile`'s related profile which is of type `ProfileNode`. So should this return `self.profile` to work correctly?

Reply:
It should not. `ProfileNode` and `YouthProfileNode` are taking their data from the same object (`YouthProfile`) which from the viewpoint of this method is `self`. A `YouthProfile` model instance does not have a field called `profile`. In the case of `ProfileNode` we just want to get the `id` of the `YouthProfile` model while the `YouthProfileNode` instance contains all fields from `YouthProfile`.

Reply:
OK, thanks for the clarification!